File: server/schemas/market_auction.py
# -*- coding: utf-8 -*-
"""경매조회 READ API 스키마 — R26-B."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict, Field


class MarketAuctionSummaryOut(BaseModel):
    model_config = ConfigDict(extra="forbid")

    auction_box_qty: str
    auction_weight_kg: str
    auction_amount: int


class MarketAuctionItemOut(BaseModel):
    model_config = ConfigDict(extra="forbid")

    trade_date: str
    auction_time: str | None = None
    market_cd: str
    market_name: str | None = None
    corporation_cd: str | None = None
    corporation_name: str | None = None
    origin_cd: str | None = None
    origin_name: str | None = None
    variety_cd: str | None = None
    variety_name: str | None = None
    unit_qty: str | None = None
    unit_nm: str | None = None
    spec_text: str | None = None
    auction_box_qty: str | None = None
    auction_weight_kg: str | None = None
    auction_price: int | None = None
    auction_amount: int = 0


class MarketAuctionListOut(BaseModel):
    model_config = ConfigDict(extra="forbid")

    trade_date: str
    fetched_at: str
    cache_hit: bool = False
    stale: bool = False
    market_cd: str
    market_name: str | None = None
    total_count: int = 0
    page: int = 1
    page_size: int = 50
    has_more: bool = False
    summary: MarketAuctionSummaryOut
    items: list[MarketAuctionItemOut] = Field(default_factory=list)

File: mobile/src/types/marketAuction.ts
/** 경매조회 READ — GET /api/v1/market-auctions (R26-B f1133eb) */

export type MarketAuctionSummary = {
  auction_box_qty: string
  auction_weight_kg: string
  auction_amount: number
}

export type MarketAuctionItem = {
  trade_date: string
  auction_time: string | null
  market_cd: string
  market_name: string | null
  corporation_cd: string | null
  corporation_name: string | null
  origin_cd: string | null
  origin_name: string | null
  variety_cd: string | null
  variety_name: string | null
  unit_qty: string | null
  unit_nm: string | null
  spec_text: string | null
  auction_box_qty: string | null
  auction_weight_kg: string | null
  auction_price: number | null
  auction_amount: number
}

export type MarketAuctionListPage = {
  trade_date: string
  fetched_at: string
  cache_hit: boolean
  stale: boolean
  market_cd: string
  market_name: string | null
  total_count: number
  page: number
  page_size: number
  has_more: boolean
  summary: MarketAuctionSummary
  items: MarketAuctionItem[]
}

export type MarketAuctionQuery = {
  trade_date: string
  market_cd?: string
  corporation_cd?: string
  origin_cd?: string // 콤마 구분 다중(OR). 미지정=전체
  variety?: string
  refresh?: boolean
  page?: number
  page_size?: number
}
